### DISCUSSION TITLE
Help with the LTO Model in

### DISCUSSION THREAD

**Discussion by bogicaa** [pybamm-team/PyBaMM#4029] [General]
Hello, 

I am currently working on my Bachelor Thesis with  the LTO Battery for ABB. I want to do the parameter optimization but had some problems when importing the parameter values from my Comsol Model into Pybamm. I don’t know if it has something to do with the functions for calculations or something else, since the datasets are for other chemistries, but even when using standard values and then manually writing that the voltage range should be from 1.7 to 2.7 V, my model in Pybamm fails to deliver this. I tried different methods that I could find on internet, but was unsuccessful until now…Could you maybe give me tips on what is the problem exactly? Even when defining cutoff voltages it falls to compute the model…The voltage range is always higher.

Best regards,

Ognjen Bogicevic

**Comment by valentinsulzer** [comment]
The "voltage limits" parameters only control the voltage cut-offs. To change the initial voltage for the simulation, try sim.solve(initial_soc="2.7V"). How well this works will depend on what parameters you have put in, especially initial concentrations.

**Comment by bogicaa** [comment]
Dear Valentin,

Thank you for your reply!
In the meantime I found out how to change this.
I changed voltage functions in Cham2020.py file so that the interpolation matches the values for LTO battery. This is working well when I tried it. I imported the rest of parameters from Comsol, also the physical ones like surface and volume of the cell but still didn’t manage to get the right curve completely… Could you maybe give me some more advanced tips? In the user manual I couldn’t find anything that could fix this.

Underneath I’m sending you my main function. So to summarize: I just changed voltage and diffusivity functions, that are working well and imported the rest of parameters from COMSOL.
When using sim.solve(initial_soc=2.7V) I get the SolverError, the solver cant find a meaningful solution. Currently the max voltage is about 2.2V, it should be around 2.7V.

Have a great day,

Ognjen Bogicevic






import pandas as pd
import numpy as np
import pybamm
import sys
import importlib.util
import pdb

module_name = "Bachelor_Thesis_2024_Basis"
module_path = r'C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project\Code Python\Bachelor_Thesis_2024_Basis.py'

spec = importlib.util.spec_from_file_location(module_name, module_path)
module = importlib.util.module_from_spec(spec)
sys.modules[module_name] = module
importlib.reload(module)
spec.loader.exec_module(module)
# Now you can use the function
custom_parameters = module.get_parameter_values()

# After loading custom parameters, check the data types
for key, value in custom_parameters.items():
    if isinstance(value, np.ndarray):  # Check if the value is a NumPy array
        print(f"{key}: Array with dtype {value.dtype}")
        if value.dtype == np.object:
            print(f"Error: {key} contains non-numeric data")
            # Attempt to convert object arrays to float, if possible
            try:
                custom_parameters[key] = value.astype(float)
            except ValueError:
                raise ValueError(f"Cannot convert {key} to float. Check input data.")

# Initialize PyBaMM's ParameterValues with your custom parameters
parameter_values = pybamm.ParameterValues(values=custom_parameters)
parameter_values = pybamm.ParameterValues("Chen2020")
parameter_values.update(custom_parameters)
print(parameter_values)

# Corrected file path
csv_file_path = r'C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project\parametersListPybamm_Corrected.csv'

# Read the CSV file
df = pd.read_csv(csv_file_path, sep=';', header=None, names=['Parameter', 'Value'])
df = df.dropna()
# Define your model
model = pybamm.lithium_ion.DFN()
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve([0,3600])
#sim.solve(initial_soc=0.5)
sim.plot()




Von: Valentin Sulzer ***@***.***>
Gesendet: Dienstag, 23. April 2024 23:10
An: pybamm-team/PyBaMM ***@***.***>
Cc: Bogicevic Ognjen ***@***.***>; Author ***@***.***>
Betreff: Re: [pybamm-team/PyBaMM] Help with the LTO Model in (Discussion #4029)


The "voltage limits" parameters only control the voltage cut-offs. To change the initial voltage for the simulation, try sim.solve(initial_soc="2.7V"). How well this works will depend on what parameters you have put in, especially initial concentrations.

**Reply by valentinsulzer** [reply]
Hard to say without seeing the parameters and data. Our user guide contains some review articles that might be helpful https://docs.pybamm.org/en/stable/source/user_guide/fundamentals/battery_models.html

**Comment by bogicaa** [comment]
I just discussed with the colleague that is also having a similar problem.

When using Chen2020 parameter set, there is no problem to fit the curve to desired voltage range, although when checking the initial concentrations, they stay the same.
When then using my own parameters from COMSOL, the SolverError occurs each time…even when changing the concentrations to values from Chen2020 or to 0, problem remains…
Enclosed I am sending you my modified code and parameters that I am using so that you can try it yourself.





import pybamm
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy.interpolate import interp1d

neg_LGM50_ocp_Chen2020_data = pybamm.parameters.process_1D_data(
    "EequNeg.csv", path=r"C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project"
)

pos_LGM50_ocp_Chen2020_data = pybamm.parameters.process_1D_data(
    "EequPos.csv", path=r"C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project"
)



def graphite_LGM50_ocp_Chen2020(sto):
    """
    LG M50 Graphite open-circuit potential as a function of stochiometry, fit taken
    from [1].

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    sto: :class:`pybamm.Symbol`
        Electrode stochiometry

    Returns
    -------
    :class:`pybamm.Symbol`
        Open-circuit potential
    """

    """
     # This part was activated before changing
    u_eq = (
        1.9793 * np.exp(-39.3631 * sto)
        + 0.2482
        - 0.0909 * np.tanh(29.8538 * (sto - 0.1234))
        - 0.04478 * np.tanh(14.9159 * (sto - 0.2769))
        - 0.0205 * np.tanh(30.4444 * (sto - 0.6103))

    )
    """


    #return(u_eq)
        #starting with the changes for eq potential
    #from scipy.interpolate import interp1d
    #import pandas as pd
    #import numpy as np


    # Load the CSV file
    csv_file_path = r'C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project\EequNeg.csv'  # Make sure to use your actual CSV file path
    df = pd.read_csv(csv_file_path, sep=',', header=None, names=['sto', 'u_eq'])

    # Convert columns to numeric, coercing errors to NaN, then drop these
    df['sto'] = pd.to_numeric(df['sto'], errors='coerce')
    df['u_eq'] = pd.to_numeric(df['u_eq'], errors='coerce')
    df.dropna(inplace=True)
    df.head()

    df = df.astype(float)
    # Assuming your CSV has two columns: 'sto' and 'u_eq'
    sto_values = df['sto'].values
    u_eq_values = df['u_eq'].values

    #voltage_values = neg_LGM50_ocp_Chen2020_data[:,1]

    # Create an interpolation function based on the values from your CSV
    if isinstance(sto, (int, float, np.ndarray)):
        # Numeric input: Perform interpolation
        u_eq_interpolation = interp1d(sto_values, u_eq_values, kind='linear', fill_value="extrapolate")
        return u_eq_interpolation(sto)
    elif isinstance(sto, pybamm.Symbol):
        # Symbolic input: Return a symbolic expression using PyBaMM functions
        # Note: Real symbolic expressions should ideally not use hardcoded numerical operations
        return pybamm.Interpolant(sto_values, u_eq_values, sto)
    else:
        raise ValueError("Unsupported type for sto. Expected numeric type or pybamm.Symbol")

    #print("Type of u_eq:", type(u_eq))
    #print("Shape of u_eq:", np.shape(u_eq))
    #_eq = float(u_eq)
    #print("Shape of u_eq:", np.shape(u_eq))
    #print("Type of u_eq:", type(u_eq))


    # Now you can use u_eq_interpolation to find u_eq for any sto value
    # For example, to find u_eq for sto = 0.5:
    #sto = 0.5

    #plotting to see how the curve will look like
    #plt.figure(figsize=(10, 5))
    #plt.plot(sto_values, u_eq_values, label='Negative Electrode OCP')
    #plt.title

#graphite_LGM50_ocp_Chen2020(1)




def graphite_LGM50_electrolyte_exchange_current_density_Chen2020(
    c_e, c_s_surf, c_s_max, T
):
    """
    Exchange-current density for Butler-Volmer reactions between graphite and LiPF6 in
    EC:DMC.

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    c_e : :class:`pybamm.Symbol`
        Electrolyte concentration [mol.m-3]
    c_s_surf : :class:`pybamm.Symbol`
        Particle concentration [mol.m-3]
    c_s_max : :class:`pybamm.Symbol`
        Maximum particle concentration [mol.m-3]
    T : :class:`pybamm.Symbol`
        Temperature [K]

    Returns
    -------
    :class:`pybamm.Symbol`
        Exchange-current density [A.m-2]
    """
    m_ref = 6.48e-7  # (A/m2)(m3/mol)**1.5 - includes ref concentrations
    E_r = 35000
    arrhenius = np.exp(E_r / pybamm.constants.R * (1 / 298.15 - 1 / T))

    return m_ref * arrhenius * c_e**0.5 * c_s_surf**0.5 * (c_s_max - c_s_surf) ** 0.5

def nmc_LGM50_ocp_Chen2020(sto):
    """
    LG M50 NMC open-circuit potential as a function of stochiometry, fit taken
    from [1].

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    sto: :class:`pybamm.Symbol`
        Electrode stochiometry

    Returns
    -------
    :class:`pybamm.Symbol`
        Open-circuit potential
    """
    """
    # part that was being taken out before changing
    u_eq = (
        -0.8090 * sto
        + 4.4875
        - 0.0428 * np.tanh(18.5138 * (sto - 0.5542))
        - 17.7326 * np.tanh(15.7890 * (sto - 0.3117))
        + 17.5842 * np.tanh(15.9308 * (sto - 0.3120))
    )

    return u_eq
    """


         #starting with the changes for eq potential
    #from scipy.interpolate import interp1d
    #import pandas as pd

    # Load the CSV file
    csv_file_path = r'C:\Users\Besitzer\OneDrive - Berner Fachhochschule\Unterricht\Battery Project\EequPos.csv'  # Make sure to use your actual CSV file path
    df = pd.read_csv(csv_file_path, sep=',', header=None, names=['sto', 'u_eq'])

    # Convert columns to numeric, coercing errors to NaN, then drop these
    df['sto'] = pd.to_numeric(df['sto'], errors='coerce')
    df['u_eq'] = pd.to_numeric(df['u_eq'], errors='coerce')
    df.dropna(inplace=True)
    df.head()

    df = df.astype(float)
    # Assuming your CSV has two columns: 'sto' and 'u_eq'
    sto_values = df['sto'].values
    u_eq_values = df['u_eq'].values

    #voltage_values = neg_LGM50_ocp_Chen2020_data[:,1]

    # Create an interpolation function based on the values from your CSV
    if isinstance(sto, (int, float, np.ndarray)):
        # Numeric input: Perform interpolation
        u_eq_interpolation = interp1d(sto_values, u_eq_values, kind='linear', fill_value="extrapolate")
        return u_eq_interpolation(sto)
    elif isinstance(sto, pybamm.Symbol):
        # Symbolic input: Return a symbolic expression using PyBaMM functions
        # Note: Real symbolic expressions should ideally not use hardcoded numerical operations
        return pybamm.Interpolant(sto_values, u_eq_values, sto)
    else:
        raise ValueError("Unsupported type for sto. Expected numeric type or pybamm.Symbol")
    #print("Type of u_eq:", type(u_eq))

    #u_eq = float(u_eq)
    #return u_eq

#nmc_LGM50_ocp_Chen2020(1)



def nmc_LGM50_electrolyte_exchange_current_density_Chen2020(c_e, c_s_surf, c_s_max, T):
    """
    Exchange-current density for Butler-Volmer reactions between NMC and LiPF6 in
    EC:DMC.

    References
    ----------
    .. [1] Chang-Hui Chen, Ferran Brosa Planella, Kieran O’Regan, Dominika Gastol, W.
    Dhammika Widanage, and Emma Kendrick. "Development of Experimental Techniques for
    Parameterization of Multi-scale Lithium-ion Battery Models." Journal of the
    Electrochemical Society 167 (2020): 080534.

    Parameters
    ----------
    c_e : :class:`pybamm.Symbol`
        Electrolyte concentration [mol.m-3]
    c_s_surf : :class:`pybamm.Symbol`
        Particle concentration [mol.m-3]
    c_s_max : :class:`pybamm.Symbol`
        Maximum particle concentration [mol.m-3]
    T : :class:`pybamm.Symbol`
        Temperature [K]

    Returns
    -------
    :class:`pybamm.Symbol`
        Exchange-current density [A.m-2]
    """

    m_ref = 3.42e-6  # (A/m2)(m3/mol)**1.5 - includes ref concentrations
    E_r = 17800
    arrhenius = np.exp(E_r / pybamm.constants.R * (1 / 298.15 - 1 / T))

    return m_ref * arrhenius * c_e**0.5 * c_s_surf**0.5 * (c_s_max - c_s_surf) ** 0.5

def electrolyte_diffusivity_Nyman2008(c_e, T):
    """
    Diffusivity of LiPF6 in EC:EMC (3:7) as a function of ion concentration. The data
    comes from [1]

    References
    ----------
    .. [1] A. Nyman, M. Behm, and G. Lindbergh, "Electrochemical characterisation and
    modelling of the mass transport phenomena in LiPF6-EC-EMC electrolyte,"
    Electrochim. Acta, vol. 53, no. 22, pp. 6356–6365, 2008.

    Parameters
    ----------
    c_e: :class:`pybamm.Symbol`
        Dimensional electrolyte concentration
    T: :class:`pybamm.Symbol`
        Dimensional temperature

    Returns
    -------
    :class:`pybamm.Symbol`
        Solid diffusivity
    """

    #D_c_e = 8.794e-11 * (c_e / 1000) ** 2 - 3.972e-10 * (c_e / 1000) + 4.862e-10
    D_c_e = 7.5e-11


    # Nyman et al. (2008) does not provide temperature dependence

    return D_c_e

def electrolyte_conductivity_Nyman2008(c_e, T):
    """
    Conductivity of LiPF6 in EC:EMC (3:7) as a function of ion concentration. The data
    comes from [1].

    References
    ----------
    .. [1] A. Nyman, M. Behm, and G. Lindbergh, "Electrochemical characterisation and
    modelling of the mass transport phenomena in LiPF6-EC-EMC electrolyte,"
    Electrochim. Acta, vol. 53, no. 22, pp. 6356–6365, 2008.

    Parameters
    ----------
    c_e: :class:`pybamm.Symbol`
        Dimensional electrolyte concentration
    T: :class:`pybamm.Symbol`
        Dimensional temperature

    Returns
    -------
    :class:`pybamm.Symbol`
        Solid diffusivity
    """

    #original funciton for conductivity

    """
    T=293
    c_e=1200
    sigma_e = (
        0.1297 * (c_e / 1000) ** 3 - 2.51 * (c_e / 1000) ** 1.5 + 3.329 * (c_e / 1000)
    )

    print(sigma_e)
    """

    #T = pybamm.Symbol('T')
    #c_e = pybamm.Symbol('c_e')

    # Set the bounds for T and c3 using PyBaMM's minimum and maximum functions
    c_ref2 = 1e6  # mol/m^3, reference concentration


    T=293
    c_e=1200
    # Define sigma_e using PyBaMM functions and symbolic expressions
    sigma_e = ((c_e / c_ref2) * (
        -10.5
        + 0.074 * (T)
        - 6.96e-5 * (T)**2
        + 668 * (c_e / c_ref2)
        - 17.8 * (c_e / c_ref2) * (T)
        + 0.028 * (c_e / c_ref2) * (T)**2
        + 4.94e5 * (c_e / c_ref2)**2
        - 886 * (c_e / c_ref2)**2 * (T)
    )**2 * 1e2) * 1  # [S/m]

    print(sigma_e)
    # Nyman et al. (2008) does not provide temperature dependence

    return sigma_e

#electrolyte_conductivity_Nyman2008(1200, 293)


# Call dict via a function to avoid errors when editing in place
def get_parameter_values():
    """
    Parameters for an LG M50 cell, from the paper :footcite:t:`Chen2020` and references
    therein.

    SEI parameters are example parameters for SEI growth from the papers
    :footcite:t:`Ramadass2004`, :footcite:t:`ploehn2004solvent`,
    :footcite:t:`single2018identifying`, :footcite:t:`safari2008multimodal`, and
    :footcite:t:`Yang2017`

    .. note::
        This parameter set does not claim to be representative of the true parameter
        values. Instead these are parameter values that were used to fit SEI models to
        observed experimental data in the referenced papers.
    """

    return {
        "chemistry": "lithium_ion",
        # sei
        "Ratio of lithium moles to SEI moles": 2.0,
        "Inner SEI reaction proportion": 0.5,
        "Inner SEI partial molar volume [m3.mol-1]": 9.585e-05,
        "Outer SEI partial molar volume [m3.mol-1]": 9.585e-05,
        "SEI reaction exchange current density [A.m-2]": 1.5e-07,
        "SEI resistivity [Ohm.m]": 200000.0,
        "Outer SEI solvent diffusivity [m2.s-1]": 2.5000000000000002e-22,
        "Bulk solvent concentration [mol.m-3]": 2636.0,
        "Inner SEI open-circuit potential [V]": 0.1,
        "Outer SEI open-circuit potential [V]": 0.8,
        "Inner SEI electron conductivity [S.m-1]": 8.95e-14,
        "Inner SEI lithium interstitial diffusivity [m2.s-1]": 1e-20,
        "Lithium interstitial reference concentration [mol.m-3]": 15.0,
        "Initial inner SEI thickness [m]": 2.5e-09,
        "Initial outer SEI thickness [m]": 2.5e-09,
        "EC initial concentration in electrolyte [mol.m-3]": 4541.0,
        "EC diffusivity [m2.s-1]": 2e-18,
        "SEI kinetic rate constant [m.s-1]": 1e-12,
        "SEI open-circuit potential [V]": 0.4,
        "SEI growth activation energy [J.mol-1]": 0.0,
        "Negative electrode reaction-driven LAM factor [m3.mol-1]": 0.0,
        "Positive electrode reaction-driven LAM factor [m3.mol-1]": 0.0,
        # cell
        "Negative current collector thickness [m]": 1.2e-05,
        "Negative electrode thickness [m]": 5e-05,            #8.52e-05,
        "Separator thickness [m]":  2.54e-05,           #1.2e-05,
        "Positive electrode thickness [m]": 3.64e-05,           #7.56e-05,
        "Positive current collector thickness [m]": 1.6e-05,
        "Electrode height [m]": 1.9416487838947598918561485112639,   #0.065,
        "Electrode width [m]":  1.9416487838947598918561485112639,              #1.58,
        "Cell cooling surface area [m2]": 0.00531,
        "Cell volume [m3]":    2.42e-05,   # 0.0005279508,        #    2.42e-05,
        "Cell thermal expansion coefficient [m.K-1]": 1.1e-06,
        "Negative current collector conductivity [S.m-1]": 58411000.0,
        "Positive current collector conductivity [S.m-1]": 36914000.0,
        "Negative current collector density [kg.m-3]": 8960.0,
        "Positive current collector density [kg.m-3]": 2700.0,
        "Negative current collector specific heat capacity [J.kg-1.K-1]": 385.0,
        "Positive current collector specific heat capacity [J.kg-1.K-1]": 897.0,
        "Negative current collector thermal conductivity [W.m-1.K-1]": 401.0,
        "Positive current collector thermal conductivity [W.m-1.K-1]": 237.0,
        "Nominal cell capacity [A.h]": 20,
        "Current function [A]": 20,
        "Contact resistance [Ohm]": 0,
        # negative electrode
        "Negative electrode conductivity [S.m-1]": 100,
        "Maximum concentration in negative electrode [mol.m-3]": 22852,
        "Negative electrode diffusivity [m2.s-1]": 2e-16,
        "Negative electrode OCP [V]": graphite_LGM50_ocp_Chen2020,
        "Negative electrode porosity": 0.25,
        "Negative electrode active material volume fraction": 0.409,  #0.409
        "Negative particle radius [m]": 1.23e-06,
        "Negative electrode Bruggeman coefficient (electrolyte)": 1.5,
        "Negative electrode Bruggeman coefficient (electrode)": 1.5,
        "Negative electrode charge transfer coefficient": 0.5,
        "Negative electrode double-layer capacity [F.m-2]": 0.2,
        "Negative electrode exchange-current density [A.m-2]"
        "":  graphite_LGM50_electrolyte_exchange_current_density_Chen2020,
        "Negative electrode density [kg.m-3]": 3400,
        "Negative electrode specific heat capacity [J.kg-1.K-1]": 700.0,
        "Negative electrode thermal conductivity [W.m-1.K-1]": 1.7,
        "Negative electrode OCP entropic change [V.K-1]": 0.0,
        # positive electrode
        "Positive electrode conductivity [S.m-1]": 3.3013,
        "Maximum concentration in positive electrode [mol.m-3]":  52060,     #52060
        "Positive electrode diffusivity [m2.s-1]": 2e-10,    #2e-10
        "Positive electrode OCP [V]": nmc_LGM50_ocp_Chen2020,
        "Positive electrode porosity": 0.335,
        "Positive electrode active material volume fraction": 0.2,   #0.2
        "Positive particle radius [m]": 2.77e-06,    #2.77e-06
        "Positive electrode Bruggeman coefficient (electrolyte)": 1.5,
        "Positive electrode Bruggeman coefficient (electrode)": 1.5,
        "Positive electrode charge transfer coefficient": 0.5,
        "Positive electrode double-layer capacity [F.m-2]": 0.2,
        "Positive electrode exchange-current density [A.m-2]"
        "": nmc_LGM50_electrolyte_exchange_current_density_Chen2020,
        "Positive electrode density [kg.m-3]": 4870,
        "Positive electrode specific heat capacity [J.kg-1.K-1]": 700.0,
        "Positive electrode thermal conductivity [W.m-1.K-1]": 2.1,
        "Positive electrode OCP entropic change [V.K-1]": 0.0,
        # separator
        "Separator porosity": 0.40,
        "Separator Bruggeman coefficient (electrolyte)": 1.5,
        "Separator density [kg.m-3]": 397.0,
        "Separator specific heat capacity [J.kg-1.K-1]": 700.0,
        "Separator thermal conductivity [W.m-1.K-1]": 0.16,
        # electrolyte
        "Initial concentration in electrolyte [mol.m-3]": 1200.0,
        "Cation transference number": 0.363,
        "Thermodynamic factor": 1.0,
        "Electrolyte diffusivity [m2.s-1]": electrolyte_diffusivity_Nyman2008,
        "Electrolyte conductivity [S.m-1]": electrolyte_conductivity_Nyman2008,
        # experiment
        "Reference temperature [K]": 298.15,
        "Total heat transfer coefficient [W.m-2.K-1]": 10.0,
        "Ambient temperature [K]": 298.15,
        "Number of electrodes connected in parallel to make a cell": 1.0,
        "Number of cells connected in series to make a battery": 1.0,
        "Lower voltage cut-off [V]": 1.7,
        "Upper voltage cut-off [V]": 2.8,
        "Open-circuit voltage at 0% SOC [V]": 1.7,
        "Open-circuit voltage at 100% SOC [V]": 2.7,
        "Initial concentration in negative electrode [mol.m-3]":  16622.0,
        "Initial concentration in positive electrode [mol.m-3]": 28046.0,
        "Initial temperature [K]": 298.15,
        # citations
        "citations": ["Chen2020"],
    }




Von: Valentin Sulzer ***@***.***>
Gesendet: Mittwoch, 24. April 2024 17:33
An: pybamm-team/PyBaMM ***@***.***>
Cc: Bogicevic Ognjen ***@***.***>; Author ***@***.***>
Betreff: Re: [pybamm-team/PyBaMM] Help with the LTO Model in (Discussion #4029)


Hard to say without seeing the parameters and data. Our user guide contains some review articles that might be helpful https://docs.pybamm.org/en/stable/source/user_guide/fundamentals/battery_models.html
